src/packet/key/public.rs

```diff
@@ -394,7 +485,24 @@ impl PublicKeyTrait for PubKeyInner {
         self.to_writer(&mut key_buf)?;
 
         // old style packet header for the key
-        writer.write_all(&[0x99, (key_buf.len() >> 8) as u8, key_buf.len() as u8])?;
+
+        match self.version {
+            KeyVersion::V2 | KeyVersion::V3 | KeyVersion::V4 => {
+                // When a v4 signature is made over a key, the hash data starts with the octet 0x99,
+                // followed by a two-octet length of the key, and then the body of the key packet.
+                writer.write_all(&[0x99, (key_buf.len() >> 8) as u8, key_buf.len() as u8])?;
+            }
+
+            KeyVersion::V6 => {
+                // When a v6 signature is made over a key, the hash data starts with the salt,
```

Comment:
where is the salt?

Reply:
In `src/packet/signature/config.rs`. But that's silly. I'll move it.

Reply:
On reflection, the existing code is right and should not be changed:

The v6 salt is part of the `SignatureConfig` and gets hashed in that top-level context. When a key is serialized for hashing (that's what we do here), the serialized key data comes after that salt.

For example, a key binding signature will hash roughly the following data:

- The salt
- Some more signature metadata from the `SignatureConfig`
- The first key (serialized with this code)
- The second key (serialized with this code)

Reply:
I (hopefully) clarified this comment